viewing_party/party.py

```diff
@@ -1,23 +1,216 @@
 # ------------- WAVE 1 --------------------
+import copy
 
 def create_movie(title, genre, rating):
-    pass
+
+    if title != None and genre != None and rating != None:
+        movie_dict = {"title":title, "genre":genre, "rating":rating}
+        return movie_dict
+    else: 
+        return None
+
+
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data    
+
+
+
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if movie["title"]==title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+
+    
+    return user_data
+
+
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    avg = 0
+    counter = 0
+    for i in user_data["watched"]:
+        for k,v in i.items():
+            if k == "rating":

```

Comment:
This loop structure will work, but it is converting a dictionary to a list and then looping to find the correct key/value pair.  An alternative is to use key in the dictionary to directly access the value:
```suggestion
        v = i["rating"]:
```
Accessing the value through the dictionary with it's key will improve readability in your code, and also make it more efficient.  Accessing keys directly is an O(1) operation, while looping over all of the keys is O(n).